single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/styles/iidm/LimitHighlightStyleProvider.java

```diff
@@ -0,0 +1,89 @@
+package com.powsybl.sld.svg.styles.iidm;
+
+import com.powsybl.iidm.network.*;
+import com.powsybl.sld.library.ComponentLibrary;
+import com.powsybl.sld.model.graphs.Graph;
+import com.powsybl.sld.model.graphs.VoltageLevelGraph;
+import com.powsybl.sld.model.nodes.*;
+import com.powsybl.sld.model.nodes.feeders.FeederWithSides;
+import com.powsybl.sld.svg.styles.EmptyStyleProvider;
+import java.util.*;
+
+import static com.powsybl.sld.svg.styles.StyleClassConstants.*;
+
+/**
+ * @author Jamal KHEYYAD {@literal <jamal.kheyyad at rte-international.com>}
+ */
+public class LimitHighlightStyleProvider extends EmptyStyleProvider {
+    Network network;
+
+    public LimitHighlightStyleProvider(Network network) {
+        this.network = network;
+    }
+
+    @Override
+    public List<String> getEdgeStyles(Graph graph, Edge edge) {
+        Optional<String> overloadStyle = getOverloadStyle(edge);
+        return overloadStyle.map(Collections::singletonList).orElse(Collections.emptyList());
+    }
+
+    private Optional<String> getOverloadStyle(Edge edge) {
+        List<Node> nodes = edge.getNodes();
+        for (Node node : nodes) {
+            if (node instanceof FeederNode feederNode) {
+                return getHighlightFeederStateStyle(feederNode);
+            } else if (node instanceof ConnectivityNode connectivityNode && hasOverloadedAdjacentNode(connectivityNode)) {
```

Comment:
Do you really need to extend the animation on the next edge? it gives a non-uniform length on the diagram, as it depends on how many switches there are on the cell. It does look a bit clumsy with those different lengths, and it is not consistent with the connection state style provider

Reply:
![first-edge](https://github.com/user-attachments/assets/c84e2ab6-0f35-4659-abf2-a3434914f28f)
![next-edge](https://github.com/user-attachments/assets/89136081-7b37-4370-bbc8-50bf058085f7)

Reply:
The product owner requested to extend the highlight to the switch. but this can be discussed

Reply:
It doesn't always extend the highlight to the switch as you're not propagating the style, you're just stopping at next edge(s):
- if there is an edge linking it to the next cell (shunt) it will blink also
- if there's no switch in this cell the blinking will stop before the busbar (because there's an internal fictitious node there)

I think we should keep it simple and similar to the connectivity state to have the feature ready for the release, and discuss back with users (or UX designer?) to see what would make more sense - and remains "good-looking" for next time.

Reply:
done. i did the change to keep it simple